packages/lesswrong/lib/components.js

```diff
@@ -132,27 +130,35 @@ import '../components/posts/PostSubmit.jsx';
 import '../components/posts/SubmitToFrontpageCheckbox';
 import '../components/posts/ReportPostMenuItem.jsx';
 
-
 import '../components/votes/VoteButton.jsx';
 import '../components/votes/CommentsVote.jsx';
 import '../components/votes/PostsVote.jsx';
 
-import '../components/localGroups/CommunityHome.jsx';
-import '../components/localGroups/CommunityMap.jsx';
-import '../components/localGroups/CommunityMapFilter.jsx';
-import '../components/localGroups/CommunityMapWrapper.jsx';
-import '../components/localGroups/EventTime.jsx';
-import '../components/localGroups/EventVicinity.jsx';
-import '../components/localGroups/LocalGroupMarker.jsx';
-import '../components/localGroups/LocalEventMarker.jsx';
-import '../components/localGroups/LocalGroupPage.jsx';
-import '../components/localGroups/LocalGroupSingle.jsx';
-import '../components/localGroups/GroupFormLink.jsx';
-import '../components/localGroups/SmallMapPreview.jsx';
-import '../components/localGroups/SmallMapPreviewWrapper.jsx';
-import '../components/localGroups/GroupLinks.jsx';
-import '../components/localGroups/LocalGroupsList.jsx';
-import '../components/localGroups/LocalGroupsItem.jsx';
+
+// events
+
+if (getSetting('hasEvents', true)) {
```

Comment:
There should be a `registerSetting` call somewhere to go with this.

Reply:
Also -- this probably works, but `import` at places other than the top level can be quite problematic in combination with Meteor's compilation. It's probably better to let these components be imported and unused, rather than conditional.

Reply:
I'm confused, when I google this I get sites saying that meteor is happy with this: https://blog.meteor.com/dynamic-imports-in-meteor-1-5-c6130419c3cd.

Reply:
Noting that this shouldn't lead to that case, will remove at your discretion.

Reply:
Pinging @jimrandomh because you had the initial comment.

Reply:
@Discordius?

Reply:
I think leaving it as is is fine. I expect we will notice any bugs quite quickly, and it's easy to reverse.